packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -1989,6 +1989,79 @@ describe('Select', () => {
                 const filteredOptions = pageObject.getFilteredOptions();
                 expect(filteredOptions.length).toBe(8);
             });
+
+            it('when clicking value, filter-input event occurs after value has been updated', async () => {
+                await clickAndWaitForOpen(element);
+                const changeEventListener = createEventListener(
+                    element,
+                    'change'
+                );
+                const filterInputEventListener = createEventListener(
+                    element,
+                    'filter-input'
+                );
+
+                const changeEventWasFirst = Promise.race([
+                    changeEventListener.promise.then(() => true),
+                    filterInputEventListener.promise.then(() => false)
+                ]);
```

Comment:
I feel like a cleaner way to do this check would be to register the same spy to both and then do assertions on the spy.calls, i.e. checking the first and second calls had expected arguments